src/thriveopedia/fossilisation/FossilisationDialog.cs

```diff
@@ -0,0 +1,232 @@
+﻿using System.Linq;
+using System.Text.RegularExpressions;
+using Godot;
+
+/// <summary>
+///   Dialog for fossilising (saving) a given species.
+/// </summary>
+public class FossilisationDialog : CustomDialog
+{
+    [Export]
+    public NodePath NameEditPath = null!;
+
+    [Export]
+    public NodePath SpeciesPreviewPath = null!;
+
+    [Export]
+    public NodePath HexPreviewPath = null!;
+
+    [Export]
+    public NodePath FossiliseButtonPath = null!;
+
+    [Export]
+    public NodePath SpeciesDetailsLabelPath = null!;
+
+    [Export]
+    public NodePath OverwriteNameConfirmationDialogPath = null!;
+
+    private LineEdit speciesNameEdit = null!;
+    private SpeciesPreview speciesPreview = null!;
+    private CellHexesPreview hexesPreview = null!;
+    private CustomRichTextLabel speciesDetailsLabel = null!;
+    private Button fossiliseButton = null!;
+    private CustomConfirmationDialog overwriteNameConfirmationDialog = null!;
+
+    /// <summary>
+    ///   The species currently open in the dialog.
+    /// </summary>
+    private Species selectedSpecies = null!;
+
+    /// <summary>
+    ///   True when one of our (name related) Controls is hovered. This needs to be known to know if a click happened
+    ///   outside the name editing controls, for detecting when the name needs to be validated.
+    /// </summary>
+    private bool controlsHoveredOver;
+
+    /// <summary>
+    ///   The species currently open in the dialog.
+    /// </summary>
+    public Species SelectedSpecies
+    {
+        get => selectedSpecies;
+        set
+        {
+            selectedSpecies = (Species)value.Clone();
```

Comment:
This clone here is a bit unexpected to me to be here, as basically always you assume a property set actually *sets* the value. Would it be possible to refactor so to keep this as-is but then create the cloned species when required?

Reply:
It's occured to me that this begs a design choice I'd implicitly answered and ignored: if a player fossilises a species with a new name, should its name change in the current game too? I added this to stop that happening, but we do have the option of using this as a method for renaming species as people have been asking for.

Reply:
I'd disallow that. The intended future feature is anyway to attach labels to things the player names, not replace their names entirely.